### DISCUSSION TITLE
I want to implement a "message discard case" in the Axum router function.

### DISCUSSION THREAD

**Discussion by kunrunic** [tokio-rs/axum#1900] [Q&A]
Hi there,

I am implementing a test server simulator.
There is a case in my implementation where I need to discard a message, but I am not sure what to return in the Axum route function. I am honestly not sure if this functionality is possible. However, I have confirmed that if I do not set a separate return value, the framework treats it as a successful request and responds accordingly.

```Rust
async fn discard(ori: OriginalUri, req: hyper::Request<Body>) {
    println!("Request URI : [{:?}]", ori);
    println!("HEADER Detail : \n{:#?}\n", req);

    let mut body = req.into_body();
    let recv = body.data().await;
    println!("DATA Detail : \n{:#?}\n", recv);

     "help me .."
}

async fn server(config: ServerConfig) {
    let st = Arc::new(config.clone());
    let app = Router::new()
        .route("/*key", post(ack).with_state(st.clone()))
        .route("/scen/discard", post(discard));
    log::info!("server run...");
    axum::Server::bind(&config.addr.parse().unwrap())
        .serve(app.into_make_service())
        .await
        .unwrap();
}
```

**Comment by jplatte** [comment]
If you panic from a handler function, hyper will by default close the client's TCP connection without returning an HTTP response. Is that what you're after?

**Reply by kunrunic** [reply]
I want to maintain the connection, but it looks difficult;;
I think it would be best to follow your advice.

As you suggested, it's okay to terminate the TCP connection without sending a response.
raised panic!() in the handler function, I confirmed that RST_STREAM was sent from the server.
However, a backtrace occurred 

I have implemented the same Test Case using Hyper's low-level code. (no backtrace occurring)
Is there a way to make axum work without backtrace?
```Rust
async fn serve(socket: TcpStream) -> Result<()> {
    let mut connection = server::handshake(socket).await?;

    while let Some(result) = connection.accept().await {
        let (request, respond) = result?;
        tokio::spawn(async move {
            if let Err(e) = handle_request(request, respond).await {
                println!("error handling request: {}", e);
            }
        });
    }

    Ok(())
}

async fn handle_request(
    mut request: Request<RecvStream>,
    mut respond: SendResponse<Bytes>,
) -> Result<()> {
    let body = request.body_mut();
    while let Some(data) = body.data().await {
        let data = data?;
        println!("<<<< recv {:?}", data);
        let _ = body.flow_control().release_capacity(data.len());
    }

    let mut response = http::Response::new(());
    if request.uri().path().ends_with("discard") == true {
        println!("discard case");
    } else {
        let bdata = b"{'KEY':'test' 'A':{}}\n";
        response
            .headers_mut()
            .insert("content-type", HeaderValue::from_static("application/json"));
        let mut send = respond.send_response(response, false)?;

        send.send_data(Bytes::from_static(b"hello "), false)?;
        send.send_data(Bytes::from_static(b"world\n"), false)?;
        send.send_data(Bytes::from_static(bdata), true)?;
    }
    Ok(())
}

```

**Reply by davidpdrsn** [reply]
I'm not sure hyper supports not sending a response without closing the connection. You're probably better off asking in the hyper repo.

**Reply by kunrunic** [reply]
It seems like we should implement the simulator case as suggested by iplatte's opinion.
Thank you for your assistance.🤩

If we find a way to discard while maintaining the connection, we will leave it in this loop. 👍
